lrlex/src/lib/lexer.rs

```diff
@@ -368,6 +368,31 @@ if 'IF'
         assert_eq!(lex2.len(), Some(2));
     }
 
+    #[test]
+    fn test_multibyte() {
+        let src = "%%
+[a❤]+ 'ID'
+[ ] ;"
+            .to_string();
+        let mut lexerdef = parse_lex(&src).unwrap();
+        let mut map = HashMap::new();
+        map.insert("ID", 0u8);
+        assert_eq!(lexerdef.set_rule_ids(&map), (None, None));
+
+        let mut lexer = lexerdef.lexer("a ❤ a");
+        let lexemes = lexer.all_lexemes().unwrap();
+        assert_eq!(lexemes.len(), 3);
+        let lex1 = lexemes[0];
+        assert_eq!(lex1.start(), 0);
+        assert_eq!(lex1.len(), Some(1));
+        let lex2 = lexemes[1];
+        assert_eq!(lex2.start(), 2);
+        assert_eq!(lex2.len(), Some(3));
```

Comment:
Would it be difficult to add another check here, that checks the content of the lexeme, just so we also test, that the lexer doesn't lex something like ` \xe2\x9d` as the second lexeme (e.g. space + the first two characters of ❤)?

Reply:
Done -- try the newest commit.

Reply:
Perfect!